docs/en/observability/monitor-aws.asciidoc

```diff
@@ -0,0 +1,596 @@
+
+[source,bash]
+----
+./filebeat test output
+----
+
+[discrete]
+==== Step 5: Configuring the {aws} Module
+
+Now that the output is working, you are going to set up the {aws} module which
+is going to automatically going to define the {aws} input. You are going to use
+the {filebeat-ref}/filebeat-module-aws.html[Amazon Web Services] module. This
+module checks SQS for new messages regarding the new object created in the S3
+bucket and uses the information in these messages to retrieve logs from S3
+buckets. With this setup, periodic polling from each S3 bucket is not needed.
+There are many diferent filesets available: `cloudtrail`, `vpcflow`, `ec2`,
+`cloudwatch`, `elb` and `s3access`. In this tutorial we are going to show you a
+few examples using the `ec2` and the `s3access` filesets. The `ec2` fileset is
+used to ship and process logs stored in cloudwatch and exported them to a S3
+bucket. The `s3access` fileset is used when S3 access logs need to be collected,
+it provides detailed records for the requests that are made to a bucket. Server
+access logs are useful for many applications. For example, access log
+information can be useful in security and access audits. It can also help you
+learn about your customer base and understand your Amazon S3 bill.
+
+Let's enable the {aws} module in {metricbeat}.
+
+[source,bash]
+----
+./filebeat modules enable aws
+----
+
+Edit the `modules.d/aws.yml` file with the following configurations.
+
+[source,yml]
+----
+- module: aws
+  cloudtrail:
+    enabled: false <1>
+  cloudwatch:
+    enabled: false <1>
+  ec2:
+    enabled: true <2>
+    var.credential_profile_name: fb-aws <3>
+    var.queue_url: https://sqs.eu-central-1.amazonaws.com/836370109380/howtoguide-tutorial <4>
+  elb:
+    enabled: false <1>
+  s3access:
+    enabled: false <1>
+  vpcflow:
+    enabled: false <1>
+----
+<1> Disables the filesets.
+<2> Enables the `ec2` fileset.
+<3> This is the aws profile define following the https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html[AWS standard].
+<4> Add the URL to the queue containing notifications around the bucket containing the EC2 logs
+
+You can now upload your logs to the S3 bucket. If you are using cloudwatch,
+make sure to edit the policy of your bucket like shown in the https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/S3ExportTasksConsole.html[step 3 of the {aws} user guide]
+it will help you avoiding permissions issues.
+
+Start {filebeat} to collect the logs.
+
+[source,bash]
+----
+./filebeat -e
+----
+
+Let's now configure another fileset and we are going to use the `s3access`
+one. The goal here is to be able to monitor how people access the bucket we
+created. This is what we achieved:
+
+image::one-bucket-archi.png[Current Architecture]
+
+Now what we need to do is to create another bucket and another queue. In order
+to create the following architecture:
+
+image::two-buckets-archi.png[Architecture with Access Logging Enabled]
```

Comment:
It seems like we're switching gears, but then we go back to what we've already achieved. What if we show the first screenshot before starting to switch context?

```suggestion
Here's what we've achieved so far:

image::one-bucket-archi.png[Current Architecture]

Now, let's configure the `s3access` fileset.
The goal here is to be able to monitor how people access the bucket we
created. To do this, we'll create another bucket and another queue.
The new architecture will look like this:

image::two-buckets-archi.png[Architecture with Access Logging Enabled]
```